compiler/model-import-verifier/src/EvalDriver.cpp

```diff
@@ -0,0 +1,161 @@
+                                                           luci::GraphBuilderSource *custom_source)
+{
+  return luci::Importer(custom_source).importModule(circle::GetModel(buffer));
+}
+
+} // namespace
+
+int entry(int argc, char **argv)
+{
+  // hardcoded values
+  const std::string conv_model_filename = "conv2d.circle";
+  const uint8_t *conv_model_const_pointer = conv2d_circle;
+  const std::vector<uint32_t> input_shape = {1, 5, 5, 1};
+  const std::vector<uint32_t> output_shape = {1, 5, 5, 2};
+  using model_dtype = float;
+
+  // size of input/output tensor in elements
+  auto const mult = std::multiplies<model_dtype>();
+  auto input_tensor_size = std::accumulate(begin(input_shape), end(input_shape), 1u, mult);
+  auto output_tensor_size = std::accumulate(begin(output_shape), end(output_shape), 1u, mult);
+
+  // allocate input buffer
+  std::vector<model_dtype> input_data(input_tensor_size);
+
+  // generate random input
+  {
+    std::random_device device;
+    std::mt19937 engine{device()};
+    std::uniform_real_distribution<model_dtype> distrib(-3, 3);
+
+    auto const generator = [&distrib, &engine]() { return distrib(engine); };
+    std::generate(begin(input_data), end(input_data), generator);
+  }
+
+  // interpret given module
+  auto const interpret_module_and_write_to_output = [&](std::unique_ptr<luci::Module> &module) {
+    // Create interpreter.
+    luci_interpreter::Interpreter interpreter(module.get());
+
+    // Set input.
+    const auto input_nodes = loco::input_nodes(module->graph());
+    assert(input_nodes.size() == 1);
+
+    auto const input_node = loco::must_cast<luci::CircleInput *>(input_nodes[0]);
+    interpreter.writeInputTensor(input_node, input_data.data(),
+                                 input_tensor_size * sizeof(model_dtype));
+
+    // Do inference.
+    interpreter.interpret();
+
+    // Get output.
+    const auto output_nodes = loco::output_nodes(module->graph());
+    assert(output_nodes.size() == 1);
+
+    auto const output_node = loco::must_cast<luci::CircleOutput *>(output_nodes[0]);
+    std::vector<model_dtype> output_data(output_tensor_size);
+    interpreter.readOutputTensor(output_node, output_data.data(),
+                                 output_tensor_size * sizeof(model_dtype));
+
+    return output_data;
+  };
+
+  // Load model from the file, import with copying, execute and save to output_buffer
+  std::vector<model_dtype> output_data_1;
+  {
+    auto module = import_model_file(conv_model_filename);
+    if (not module)
+      throw std::runtime_error("Fail to import model");
+
+    output_data_1 = interpret_module_and_write_to_output(module);
+  }
+
+  // Load model from const pointer, import with copying, execute and save to output_buffer
+  std::vector<model_dtype> output_data_2;
+  {
+    // default builder source not allows using constants from model's buffer
+    auto default_source = &luci::GraphBuilderRegistry::get();
```

Comment:
Thank you for usage example!
I think you can highlight this part in Notes, so it will be easier to find it in this PR (it is large :slightly_smiling_face: )

Reply:
As far I understand your suggestion I did that.